packages/server/src/next-utils.ts

```diff
@@ -65,13 +77,31 @@ export function nextBuild(nextBin: string, cwd: string) {
 }
 
 export async function nextStart(nextBin: string, cwd: string, config: ServerConfig) {
-  const availablePort = await detect({port: config.port, hostname: config.hostname})
-  return Promise.resolve(
-    spawn(nextBin, ['start', '-p', `${availablePort}`, '-H', config.hostname], {
-      cwd,
-      stdio: 'inherit',
-    }).on('error', (err) => {
-      console.error(err)
-    }),
-  )
+  let spawnCommand: string[] = ['start']
+  let availablePort: number
+  if (config.port) {
+    availablePort = await detect({port: config.port})
+    spawnCommand = spawnCommand.concat(['-p', `${config.port}`])
+  }
+  if (config.hostname) {
+    spawnCommand = spawnCommand.concat(['-H', `${config.hostname}`])
+  }
```

Comment:
There is an optional opportunity here to consolidate adding the port and hostname into a function that is shared between the start and dev functions.